### PR TITLE
[HOPSWORKS-2014] assume role docs

### PR DESCRIPTION
**Pull request by ErmiasG**
logicalclocks/hops-testing-ee/pull/197

### CODE REVIEW

File: docs/admin_guide/cloud_role_mapping.rst [hunk 1]
```diff
@@ -0,0 +1,95 @@
+
+==================
+AWS Role chaining
+==================
+
+Using an EC2 instance profile you can allow, your hopsworks cluster, access to AWS resources. This forces all hopsworks users to
```
Comment (tkakantousis) [inline docs/admin_guide/cloud_role_mapping.rst:6]:
```suggestion
Using an EC2 instance profile enables your Hopsworks cluster to access AWS resources. This forces all Hopsworks users to
```
```suggestion
Using an EC2 instance profile you can allow, your hopsworks cluster, access to AWS resources. This forces all hopsworks users to
```

File: docs/admin_guide/cloud_role_mapping.rst [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+share the instance profile role and the resource access policies attached to that role. To allow for per project access policies 
+you could have your users use AWS credentials directly in their programs which is not recommended or you can and should use
```
Comment (tkakantousis) [inline docs/admin_guide/cloud_role_mapping.rst:8]:
```suggestion
you could have your users use AWS credentials directly in their programs which is not recommended and should use
```
```suggestion
you could have your users use AWS credentials directly in their programs which is not recommended or you can and should use
```

File: docs/admin_guide/cloud_role_mapping.rst [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+`Role chaining <https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_terms-and-concepts.html#iam-term-role-chaining>`_.
+
+To use Role chaining, you need to first setup IAM roles in AWS:
+
+1. Create an instance profile role with policies that will allow it to assume all resource roles that we can assume from the hopsworks
```
Comment (tkakantousis) [inline docs/admin_guide/cloud_role_mapping.rst:13]:
```suggestion
1. Create an instance profile role with policies that will allow it to assume all resource roles that we can assume from the Hopsworks
```
```suggestion
1. Create an instance profile role with policies that will allow it to assume all resource roles that we can assume from the hopsworks
```

File: docs/admin_guide/cloud_role_mapping.rst [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+                "AWS": "arn:aws:iam::xxxxxxxxxxxx:role/instance-profile"
+            },
+            "Action": "sts:AssumeRole"
+            }
+        ]
+    }
+
+Example policy document.
+
+3. Finally attache the instance profile to the master node of you hopsworks EC2 instances.
```
Comment (tkakantousis) [inline docs/admin_guide/cloud_role_mapping.rst:56]:
```suggestion
3. Finally attach the instance profile to the master node of your Hopsworks AWS instance.
```
```suggestion
3. Finally attache the instance profile to the master node of you hopsworks EC2 instances.
```

File: docs/admin_guide/cloud_role_mapping.rst [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+.. _project-roles.png: ../_images/admin/project-roles.png
+.. figure:: ../imgs/admin/project-roles.png
+  :alt: Project roles
+  :target: `project-roles.png`_
+  :align: center
+  :figclass: align-cente
+ 
+  Cloud roles mapped to project.
+
+For instractions on how to use the assume role API go to :doc:`../user_guide/hopsworks/assume-role`
```
Comment (tkakantousis) [inline docs/admin_guide/cloud_role_mapping.rst:95]:
```suggestion
For instructions on how to use the assume role API go to :doc:`../user_guide/hopsworks/assume-role`
```
```suggestion
For instractions on how to use the assume role API go to :doc:`../user_guide/hopsworks/assume-role`
```

File: docs/hopsml/experiment.rst [hunk 6]
```diff
@@ -72,6 +72,19 @@ The path returned is to the root directory in Hopsworks of your project.
 
     ... Experiment code ...
 
+**Step 2 (optional)**. If you have your data in an s3 bucket you can read the data using TensorFlow by
```
Comment (tkakantousis) [inline docs/hopsml/experiment.rst:75]: That is not required if the instance has a role that can read from S3, is that right? Maybe we could clarify that only if the instance does not have the role, they need to set the environment variables. 

Also isn't the experiments using hops-util-py which means that these variables are already set?

File: docs/user_guide/hopsworks/assume-role.rst [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+.. _project-roles.png: ../_images/admin/project-roles.png
+.. figure:: ../../imgs/admin/project-roles.png
+ :alt: Project roles
+ :target: `project-roles.png`_
+ :align: center
+ :figclass: align-cente
+
+ Cloud roles mapped to project.
+ 
+You can then use the Hops Util python and java APIs to assume the roles listed in your project settings page.
```
Comment (tkakantousis) [inline docs/user_guide/hopsworks/assume-role.rst:21]:
```suggestion
You can then use the Hops Python and Java APIs to assume the roles listed in your project's settings page.
```

File: docs/user_guide/hopsworks/assume-role.rst [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+ 
+ 
+.. code-block:: scala
+ 
+   import io.hops.util.CredentialsProvider
+   val creds = CredentialsProvider.assumeRole(CredentialsProvider.getRole(1))
+   spark.read.csv("s3a://resource/test.csv").show()
+ 
+The assume role method sets spark hadoop configurations that will allow spark to read s3 buckets. The code
+examples above show how to read s3 buckets using python and scala.
```
Comment (tkakantousis) [inline docs/user_guide/hopsworks/assume-role.rst:40]:
```suggestion
examples above show how to read s3 buckets using Python and Scala.
```

**Review by tkakantousis — CHANGES_REQUESTED** [review]

File: docs/admin_guide/cloud_role_mapping.rst (hunk 2 above)
Comment (ErmiasG) [reply docs/admin_guide/cloud_role_mapping.rst:8]: I do not think the sentence makes sense if I remove or

File: docs/hopsml/experiment.rst (hunk 6 above)
Comment (ErmiasG) [reply docs/hopsml/experiment.rst:75]: But we said you should not use a single instance profile role. 

What do you mean by experiments using hops-util-py which means that these variables are already set?

**Review by tkakantousis — APPROVED** [review]

**Pull request merged**